### ISSUE TITLE
Input Symbol only works from Cmd+Shift+P

### ISSUE THREAD

**Issue by TheoWinterhalter** [banacorn/agda-mode#10]
I hope I am not missing something, but when I just type `\`, it does not realize I want to insert a symbol.
I need to open the palette to do so.

**Comment by banacorn** [comment]
What do you mean by a palette? The panel that pops up after your loaded the file (`ctrl-c ctrl-l`) ?
The input method activates only when the file extension is `.agda`

**Comment by iSheeft** [comment]
Sorry I wasn't clear enough.
When I do `Cmd + Shift + P`, select Agda Mode: Input Symbol, and then type `bn`, it displays `ℕ`, but when I just type `\bn`, nothing happens (even after `^C ^L`).

**Comment by banacorn** [comment]
This is what the keybinding looks like on my editor,  typing `\` alone should be enough to invoke the input method. 

![screensht](http://i.imgur.com/CO50C0H.png)

Could you please share what your keybinding looks like?

**Comment by iSheeft** [comment]
As you can see, this is the same for me.
![capture d ecran 2015-05-08 a 17 02 52](https://cloud.githubusercontent.com/assets/5960860/7538984/2a5d3ce8-f5a4-11e4-9abc-87448da2a6ca.png)

I am on Mac OS X Yosemite 10.10.4 (14E11f) using the last version of Atom and of your package.

**Comment by banacorn** [comment]
Does it pops that next symbol suggestion thingy when you simply type `\`?

**Comment by iSheeft** [comment]
Nope, it only does that with `Cmd-Shift-P`.

I have one warning related to your package, but I am not sure it had something to do with the bug at hand (which you don't have apparently?).

![capture d ecran 2015-05-09 a 22 52 08](https://cloud.githubusercontent.com/assets/5960860/7552001/250f290e-f69e-11e4-9085-cca68461188d.png)

**Comment by banacorn** [comment]
I've send a patch, please update to the latest version and see if the problem still persists

**Comment by iSheeft** [comment]
Thanks, but I am afraid nothing have changed (except that you have fixed all the warnings).

**Comment by iSheeft** [comment]
I just updated to the last version of your package, and it still doesn't work.
Am I the only one having this problem?

**Comment by banacorn** [comment]
Does other commands also have this problem, or is it just `agda-mode:input-symbol` ?

Is it possible that the keystroke got overridden by other packages, which also occupies `\`

**Comment by iSheeft** [comment]
No, I have tested them, and they seem to work just fine.

Apparently there is no keystroke on `\` (but writing the `\` itself of course), just one on `cmd-\`:
![capture d ecran 2015-05-26 a 22 37 41](https://cloud.githubusercontent.com/assets/5960860/7822776/147b586c-03f8-11e5-8408-765cbef545f1.png)

**Comment by iSheeft** [comment]
Even better:

```cson
'atom-text-editor.agda':
  'alt-/': 'agda-mode:input-symbol'
```

This is the same combination as for `\` and it actually works.
Thanks.

**Comment by banacorn** [comment]
Thanks!

Does it has something to do with keyboard layouts? (localization, not physical layout)
https://github.com/atom/atom-keymap/issues/35
https://github.com/atom/atom/issues/1625

`\\` works fine for me and some of the others while `alt-/` doesn't
But it's okay we will add both keystrokes to the keymap anyway!

**Comment by iSheeft** [comment]
I don't think so, but it is hard to tell.
Using the right alt doesn't change anything for me (it works with `alt-/` and doesn't with `\\`).

**Comment by banacorn** [comment]
It's been a while, and the codebase has changed a lot since then, is this still an issue?

**Comment by TheoWinterhalter** [comment]
Hi. It worked last time and it still does.
I think it's because of the trick we mentioned earlier but it's not that bad is it?

It's true we could have closed the issue, I'll leave it to you.

**Comment by banacorn** [comment]
I'm leaving this open as long as `\` doesn't work

**Comment by TheoWinterhalter** [comment]
Maybe I forgot something but I don't know how to remove the key binding `alt-/` to test.

**Comment by banacorn** [comment]
Oh, I added `alt-/` on purpose dated back when you said it's the same combination as for `\`

**Comment by TheoWinterhalter** [comment]
Yes I do know that. I just don't know how to test if `\` is now recognized on its own without this trick.

**Comment by amnipar** [comment]
I just installed the newest atom-beta, language-agda and agda-mode and the problem persists. I have a Finnish keyboard layout, this might have something to do with the problem. When I Alt-press some keys, the key binding resolver gives me wrong keys for many non-letter keys and alt-shift-number combinations. The strange thing is, that when I press the key alone or shift-number alone, the correct character is printed, according to my keyboard layout. But for some reason the key binding resolver gets the wrong key from some other layout. As an example, I found out that Alt-' on my keyboard corresponds to Alt-/, so I can use that to enable agda character input mode, so I have a solution to the problem. However, this issue is not resolved since somehow the key binding resolver gets wrong keys. I have no idea where to start looking for the problem. I compiled from source and have the git repository cloned, so if anyone has any idea, I can give it a try.

**Comment by banacorn** [comment]
Thank you for reporting this!

Do you have any other [keyboard functional layouts](https://en.wikipedia.org/wiki/Keyboard_layout), aside from the Finnish one?

I have a Taiwanese keyboard, with a visual layout labeled with exotic symbols:
![Taiwanese keyboard visual layout](https://support.apple.com/library/content/dam/edam/applecare/images/en_US/keyboards/taiwanese_notebook.png)

There are 2 functional layouts I'm using: Taiwanese and US. I do everything with the US layout (including coding), and I switch to the Taiwanese layout only when I'm typing Chinese characters. 

**When I'm using the US layout, everything works fine. But I get weird symbols when I `Alt-` press every key with the Taiwanese layout.**

Perhaps switching to the US functional layout would solve the problem, or we could also add another combination of keys that doesn't get in the way with the Finnish layout. :smile:

**Comment by amnipar** [comment]
I only have the Finnish keyboard. I investigated some more, and it seems that with Alt-<key>, the <key> is interpreted as if my keyboard had US layout. Without Alt, I get keys correctly as in Finnish layout. Very strange. I can work around the problem, but it would be nice to figure out why this happens.

**Comment by banacorn** [comment]
Maybe it would be a good idea to install another functional layout (input method). 

I get to switch my  keyboard layout (input method) between the US and TW with a key combination (⌘ + space), which frees me from having to hold on to the `alt` key or something.
![switching input method](http://i.imgur.com/WHv1GTJ.png)

**Comment by rjstone** [comment]
The plain old `\` unicode sequences are working for me (us eng kbd), eg `\bN` types the nat N, etc

**Comment by banacorn** [comment]
I'm closing this because I've rewritten the whole input system.
Please reopen this issue if the problem still persists.
